light-client/src/consensus/consensus_client.rs

```diff
@@ -588,6 +475,124 @@ impl<R: ConsensusRpc> ConsensusLightClient<R> {
+        && update.signature_slot > update.attested_header.slot
+        && update.attested_header.slot >= update_finalized_slot;
+
+    if !valid_time {
+        return Err(ConsensusError::InvalidTimestamp.into());
+    }
+
+    let store_period = calc_sync_period(store.finalized_header.slot);
+    let update_sig_period = calc_sync_period(update.signature_slot);
+    let valid_period = if store.next_sync_committee.is_some() {
+        update_sig_period == store_period || update_sig_period == store_period + 1
+    } else {
+        update_sig_period == store_period
+    };
+
+    if !valid_period {
+        return Err(ConsensusError::InvalidPeriod.into());
+    }
+
+    let update_attested_period = calc_sync_period(update.attested_header.slot);
+    let update_has_next_committee = store.next_sync_committee.is_none()
+        && update.next_sync_committee.is_some()
+        && update_attested_period == store_period;
+
+    if update.attested_header.slot <= store.finalized_header.slot && !update_has_next_committee {
+        return Err(ConsensusError::NotRelevant.into());
+    }
+
+    if update.finalized_header.is_some() && update.finality_branch.is_some() {
+        let is_valid = is_finality_proof_valid(
+            &update.attested_header,
+            &mut update
+                .finalized_header
+                .clone()
+                .expect("we know that this is `Some`"),
+            &update
+                .finality_branch
+                .clone()
+                .expect("we know that this is `Some`"),
+        );
+
+        if !is_valid {
+            return Err(ConsensusError::InvalidFinalityProof.into());
+        }
+    }
+
+    if update.next_sync_committee.is_some() && update.next_sync_committee_branch.is_some() {
+        let is_valid = is_next_committee_proof_valid(
+            &update.attested_header,
+            &mut update
+                .next_sync_committee
+                .clone()
+                .expect("we know that this is `Some`"),
+            &update
+                .next_sync_committee_branch
+                .clone()
+                .expect("we know that this is `Some`"),
+        );
+
+        if !is_valid {
+            return Err(ConsensusError::InvalidNextSyncCommitteeProof.into());
+        }
+    }
+
+    let sync_committee = if update_sig_period == store_period {
+        &store.current_sync_committee
+    } else {
+        store
+            .next_sync_committee
+            .as_ref()
+            .expect("we know that this is `Some` because we are in `valid_period`")
+    };
+
+    let pks = get_participating_keys(sync_committee, &update.sync_aggregate.sync_committee_bits)?;
+
+    let is_valid_sig = verify_sync_committee_signature(
+        &pks,
+        &update.attested_header,
+        &update.sync_aggregate.sync_committee_signature,
+        genesis_root,
+        fork_version,
+    );
+
+    if !is_valid_sig {
+        return Err(ConsensusError::InvalidSignature.into());
+    }
```

Comment:
```suggestion
    ensure!(is_valid_signature, ConsensusError::InvalidSignature.into());
```

nit: This seems cleaner